lib/ath.groovy

```
@@ -0,0 +1,125 @@
+        }
+    }
+
+    branches['ath-split-cucumber'] = {
+        this.singleAthBranch(functions, null, mvnFlags + " -Dcucumber.test=features", 'split-cucumber', nodeLabel, archiveJUnitReports)
+    }
+    return branches;
+}
+
+void singleAthBranch(def functions,
+                     def exclusions,
+                     String mvnProps,
+                     String splitId,
+                     String nodeLabel,
+                     boolean archiveJUnitReports = false) {
+    if (nodeLabel == null) {
+        nodeLabel = athLabel()
+    }
+    
+    node(nodeLabel) {
+        unstash 'ath-stash'
+        unstash 'jenkins.war'
+        if (exclusions != null) {
+            def excludeStr = exclusions.join("\n")
+            writeFile file: 'excludes.txt', text: excludeStr
+            echo excludeStr
+        }
+        timeout(time: 6, unit: TimeUnit.HOURS) { //TODO might need trimming
+            wrap([$class: 'Xvnc', takeScreenshot: false, useXauthority: true]) {
+                functions.withMaven {
+                    sh 'mvn clean test -B -Dmaven.test.failure.ignore=true -DforkCount=1 ' + mvnProps
+                }
+            }
+            try {
+                step([$class: 'JUnitResultArchiver', testResults: 'target/surefire-reports/*.xml',
+                     testDataPublishers: [[$class: 'AttachmentPublisher']]])
+
+                if (archiveJUnitReports) {
+                    //todo use zip step
+                    sh "find . -path \"*/target/surefire-reports/*.xml\" | zip -@ junitReports-${splitId}.zip"
+                    archive "junitReports-${splitId}.zip"
+                }
+            } catch (Exception ignored) {
+                echo "No test reports found."
+            }
+            try {
+                archive "**/target/diagnostics/**"
+            } catch (Exception e) {
+                echo "No diagnostics found."
```

If we expect the archive to be empty we should instruct the archiver to allow it instead of swallowing whatever exception it throws.